src/cluster_legacy.c

```diff
@@ -3121,6 +3122,18 @@ int clusterProcessPacket(clusterLink *link) {
         return 1;
     }
 
+    if (type == CLUSTERMSG_TYPE_MEET && link->node && nodeInHandshake(link->node)) {
```

Comment:
Instead of dropping it, should we should skip creating the new node down where we are normally handling new meet messages?

Reply:
It feels like if we are in handshake with the other node, and we get a meet message, we should be OK to put the node online? My understanding is that the bug is more around the double setup around the node.

Reply:
What are the risks of dropping it (if it's in a short span of time) ?

Reply:
Does't this condition mean that we would never end up entering this other condition now?
https://github.com/valkey-io/valkey/blob/5f7fe9ef21f1feae42257ac93ab33d8f8c06e97f/src/cluster_legacy.c#L3276-L3292

Reply:
> What are the risks of dropping it (if it's in a short span of time) ?

If an end user sends a `cluster meet` from both nodes at the same time, wouldn't this trigger it to drop the meet packet unnecessarily? I think that might be fine, it just seems not correct to drop the packet.

Reply:
i think we should somehow to drop the MEET or go through the code and find out if we want to keep the MEET, what else code should we do. Droping it seem a safer way i guess, but if we can keep the MEET and make it work, i guess it is a better way.